pkg/transaction/transaction.go

```diff
@@ -274,23 +276,25 @@ func (t *transactionService) prepareTransaction(ctx context.Context, request *Tx
 
 	gasPrice := request.GasPrice
 	if gasPrice == nil {
-		gasPriceSuggested, err := t.backend.SuggestGasPrice(ctx)
-		gasPrice = new(big.Int).Div(new(big.Int).Mul(big.NewInt(int64(boostPercent)+100), gasPriceSuggested), big.NewInt(100))
+		gasPrice, err = t.backend.SuggestGasPrice(ctx)
 		if err != nil {
 			return nil, err
 		}
 	}
+	gasPrice = new(big.Int).Div(new(big.Int).Mul(big.NewInt(int64(tipPercent)+100), gasPrice), big.NewInt(100))
 	if gasPrice.Cmp(minGasPrice) < 0 {
 		return nil, ErrGasPriceTooLow
 	}
 
-	return types.NewTx(&types.LegacyTx{
-		Nonce:    nonce,
-		To:       request.To,
-		Value:    request.Value,
-		Gas:      gasLimit,
-		GasPrice: gasPrice,
-		Data:     request.Data,
+	return types.NewTx(&types.DynamicFeeTx{
+		Nonce:     nonce,
+		ChainID:   t.chainID,
+		To:        request.To,
+		Value:     request.Value,
+		Gas:       gasLimit,
+		GasTipCap: gasPrice,
+		GasFeeCap: gasPrice,
```

Comment:
they both use the same value?

Reply:
yes, I've added some comments that clarifies it hopefully